sopel/tools/jobs.py

```diff
@@ -54,20 +99,45 @@ def stop(self):
 
         The scheduler thread will stop its loop over jobs to process, but it
         won't join the thread, or clear its queue—this has to be done
-        separately by the calling thread.
+        separately by the calling thread::
+
+            scheduler.stop()  # ask the scheduler to stop
+            scheduler.join()  # wait for the scheduler to actually stop
+
+        Note that this won't cancel or stop any currently running jobs.
         """
         self.stopping.set()
 
     def remove_callable_job(self, callable):
-        """Remove ``callable`` from the job queue."""
+        """Remove ``callable`` from the job queue.
+
+        :param callable callable: the callable to remove
+        :type callable: :term:`function`
+
+        This method is thread safe. However, it won't cancel or stop any
+        currently running jobs.
+        """
         with self._mutex:
             self._jobs = [
                 job for job in self._jobs
-                if job.func != callable
+                if job._handler != callable
             ]
 
     def run(self):
-        """Run forever until :attr:`stopping` event is set."""
+        """Run forever until :meth:`stop` is called.
+
+        This method waits at most a second between each iteration. At each step
+        it retrieves the jobs that are ready for execution, and, well, execute
+        them. See the :meth:`Job.execute` method for more information.
```

Comment:
```suggestion
        it retrieves the jobs that are ready for execution, and executes them.
        See the :meth:`Job.execute` method for more information.
```

I do appreciate the bit of cheek, but don't think it belongs in this kind of documentation. :/